shared-module/audiodelays/Echo.c

```diff
@@ -467,12 +474,12 @@ audioio_get_buffer_result_t audiodelays_echo_get_buffer(audiodelays_echo_obj_t *
                     word = echo + sample_word;
```

Comment:
If you happen to do a commit before the final review and have a moment after this line could you add:
`word = mix_down_sample(word);`
I'll try to make it more elegant later but there is an issue and that fixes it.

So the final chunk is:
```
word = echo + sample_word;
word = mix_down_sample(word);

Reply:
```
shared-module/audiodelays/Echo.c:int16_t mix_down_sample(int32_t sample) {
shared-module/audiofilters/Filter.c:int16_t mix_down_sample(int32_t sample) {
shared-module/synthio/__init__.c:int16_t mix_down_sample(int32_t sample) {
```
we've now got 3 copies of mix_down_sample. If not as part of this PR, please make this have extern linkage somewhere and get rid of the duplicates.